src/utils.rs

```diff
@@ -0,0 +1,25 @@
+//! Utilities for consumers of the raw API
+use vst3_com::{ComInterface, ComRc};
+
+/// A thin wrapper around a raw pointer to a vtable. Used in traits that return pointers to instances.
+#[repr(transparent)]
+#[derive(Copy, Clone, Debug)]
+pub struct VstPtr<I: ComInterface + ?Sized> {
+    inst: *mut *mut <I as ComInterface>::VTable,
+}
+
+impl<I: ComInterface + ?Sized> VstPtr<I> {
+    /// check if the underlying pointer is null
+    pub fn is_null(&self) -> bool {
+        self.inst.is_null()
+    }
+    /// Promote the pointer to a reference count, returns `None` if the pointer is null.
+    pub fn upgrade(&self) -> Option<ComRc<I>> {
```

Comment:
Here we need to replace ComRc with ComPtr to fix the segfault

Reply:
@MirkoCovizzi fixed